File: xinya-backend/src/main/java/com/xinya/dtx/dto/EducationDto.java
package com.xinya.dtx.dto;

import com.xinya.dtx.entity.EducationContent;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Arrays;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class EducationDto {
    private String id;
    private String title;
    private String category;
    private String description;
    private String contentType;
    private Integer durationSeconds;
    private String thumbnailUrl;
    private String mediaUrl;
    private List<String> tags;
    
    public static EducationDto fromEntity(EducationContent content) {
        return EducationDto.builder()
            .id(content.getId())
            .title(content.getTitle())
            .category(content.getCategory())
            .description(content.getDescription())
            .contentType(content.getContentType())
            .durationSeconds(content.getDurationSeconds())
            .thumbnailUrl(content.getThumbnailUrl())
            .mediaUrl(content.getMediaUrl())
            .tags(content.getTags() != null ? 
                  Arrays.asList(content.getTags().split(",")) : List.of())
            .build();
    }
}

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
class EducationListResponse {
    private List<EducationDto> contents;
    private Integer total;
}

File: xinya-client/app/src/main/java/com/xinya/dtx/core/network/dto/EducationDto.kt
package com.xinya.dtx.core.network.dto

import com.google.gson.annotations.SerializedName

/**
 * 宣教内容DTO
 */
data class EducationContentDto(
    val id: String,
    val title: String,
    val category: String,
    val description: String,
    @SerializedName("content_type")
    val contentType: String,            // "video" | "article"
    @SerializedName("duration_seconds")
    val durationSeconds: Int,
    @SerializedName("thumbnail_url")
    val thumbnailUrl: String?,
    @SerializedName("media_url")
    val mediaUrl: String?,
    val tags: List<String> = emptyList()
)

/**
 * 宣教内容列表响应
 */
data class EducationListResponse(
    val contents: List<EducationContentDto>,
    val total: Int
)
